flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1721,6 +1721,25 @@ bool OmpAttributeVisitor::Pre(const parser::DoConstruct &x) {
           }
         }
       }
+    } else if (x.IsDoConcurrent()) {
+      const Fortran::parser::LoopControl *loopControl = &*x.GetLoopControl();
+      const Fortran::parser::LoopControl::Concurrent &concurrent =
+          std::get<Fortran::parser::LoopControl::Concurrent>(loopControl->u);
+      const Fortran::parser::ConcurrentHeader &concurrentHeader =
+          std::get<Fortran::parser::ConcurrentHeader>(concurrent.t);
+      const std::list<Fortran::parser::ConcurrentControl> &controls =
+          std::get<std::list<Fortran::parser::ConcurrentControl>>(
+              concurrentHeader.t);
+      for (const auto &control : controls) {
+        const parser::Name *iv{&std::get<0>(control.t)};
+        if (iv && iv->symbol) {
+          if (!iv->symbol->test(Symbol::Flag::OmpPreDetermined)) {
+            ResolveSeqLoopIndexInParallelOrTaskConstruct(*iv);
+          } else {
+            // TODO: conflict checks with explicitly determined DSA
+          }
+        }
+      }
```

Comment:
Is the threadprivate check not applicable here?

Reply:
**The loop iteration variable may not appear in a threadprivatedirective**. Yeah, sorry I missed it. If it's okay, I'll store symbols in a list and move common code out.

Reply:
Ok

Reply:
Due to a new symbol being declared [here](https://github.com/llvm/llvm-project/blob/394e321a2ea021194891cf5b0c5747a901fb7915/flang/lib/Semantics/resolve-names.cpp#L6865), the check won't work for DoConcurrent. I'll create an issue to try to handle it in a separate patch.